### PR TITLE
Fixed & Added scripts

### PR DESCRIPTION
**Pull request by kmsiapps**
* Fixed broken spawn_protect.py (Fixed typo)

* Added Spade-nade bug block script(spadenadefix.py)
(If you held spade and change tool into grenades with RMB clicked, the grenade would throw and explode instantly. This script fixes it)

### CODE REVIEW

File: feature_server/config/scripts/spadenadefix.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""
+spadenadefix.py - blocks spade-nade bug
+by kmsi(kmsiapps@gmail.com)
+version 1(2017.01.21)
+"""
+def apply_script(protocol,connection,config):
+
+    class kmsiConnection(connection):
```
Comment (noway421) [inline feature_server/config/scripts/spadenadefix.py]: Would be better to use established naming convention in the form of "${scriptname}Connection"

Change it to `SpadenadeFixConnection` in that case?

File: feature_server/config/scripts/spawn_protect.py [hunk 2]
```diff
@@ -2,30 +2,28 @@
...
+Fixed error by kmsi(kmsiapps@gmail.com) : replaced player to hit_player
 """
 
 from pyspades.common import prettify_timespan
 from twisted.internet import reactor
 
 def apply_script(protocol, connection, config):
     spawn_protect_time = config.get('spawn_protect_time', 3.0)
-
+
```
Comment (noway421) [inline feature_server/config/scripts/spawn_protect.py]: Please don't leave trailing white space

File: feature_server/config/scripts/spawn_protect.py [hunk 3]
```diff
@@ -2,30 +2,28 @@
...
-                "%s is spawn-protected for %s." %
-                    (player.name,
-                     prettify_timespan(timespan, True)))
+                self.send_chat( "%s is spawn-protected for %s." %(hit_player.name, prettify_timespan(timespan, True)))
                 return False
-            return connection.on_hit(self, hit_amount, player, type, grenade)
-
-    return protocol, SpawnProtectConnection
+            return connection.on_hit(self, hit_amount, hit_player, type, grenade)
+
```
Comment (noway421) [inline feature_server/config/scripts/spawn_protect.py]: Please remove trailing white space

File: feature_server/config/scripts/spawn_protect.py [hunk 4]
```diff
@@ -2,30 +2,28 @@
...
-                    (player.name,
-                     prettify_timespan(timespan, True)))
+                self.send_chat( "%s is spawn-protected for %s." %(hit_player.name, prettify_timespan(timespan, True)))
                 return False
-            return connection.on_hit(self, hit_amount, player, type, grenade)
-
-    return protocol, SpawnProtectConnection
+            return connection.on_hit(self, hit_amount, hit_player, type, grenade)
+    
+    return protocol, SpawnProtectConnection
```
Comment (feikname) [inline feature_server/config/scripts/spawn_protect.py]: The final line doesn't end with a EOL here.

File: feature_server/config/scripts/spadenadefix.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+"""
+spadenadefix.py - blocks spade-nade bug
+by kmsi(kmsiapps@gmail.com)
+version 1(2017.01.21)
+"""
+def apply_script(protocol,connection,config):
```
Comment (feikname) [inline feature_server/config/scripts/spadenadefix.py]: Could you please add spaces after the commas? To improve readability :)

File: feature_server/config/scripts/spawn_protect.py (hunk 4 above)
Comment (kmsiapps) [inline feature_server/config/scripts/spawn_protect.py]: @feikname Mmm.. I don't know what are you meaning. Can you explain it more easily?

**Comment by kmsiapps** [comment]
* Removed trailing spaces, changed protocol name into Spadenadeconnection
* Added nosharp.py - It would hopefully fix this problem : https://github.com/piqueserver/piqueserver/issues/3

Needed some test, although I made some test before commit.

File: feature_server/config/scripts/spawn_protect.py (hunk 4 above)
Comment (noway421) [inline feature_server/config/scripts/spawn_protect.py]: An empty line as the last line of the file. Not all text editors insert it automatically, although they should. Right now it ends abrubtly.

Basically you know you got it right when you get rid of this little icon in github file view: 
screen shot 2017-01-22 at 8 04 34 pm

Comment (kmsiapps) [inline feature_server/config/scripts/spawn_protect.py]: @noway421 Aha, I got it. thanks for the kind explanation!

Comment (feikname) [inline feature_server/config/scripts/spawn_protect.py]: @kmsiapps [this StackOverflow question](http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) explains it quite well.

Basically, some editors won't place the End-of-line (*[EOL](https://en.wikipedia.org/wiki/Newline)*) character at the end of the last line unless you add an additional one below. However, that EOL character is necessary for some programs to work correctly with the file. 

noway421 tip about getting rid of the little icon is pretty good too :)

Comment (kmsiapps) [inline feature_server/config/scripts/spawn_protect.py]: @feikname I updated my scripts again. Is it okay now?

**Comment by feikname** [comment]
It's very common to have a bunch of commits :P

**Comment by noway421** [comment]
Looks good to me :+1:

**Pull request merged**